pipenv/utils/resolver.py

```diff
@@ -210,14 +212,14 @@ def create(
                 pipfile_entries[package_name] = pipfile_entry
                 if isinstance(pipfile_entry, dict):
                     if packages[package_name].get("index"):
-                        index_lookup[package_name] = packages[package_name].get("index")
+                        index_lookup[canonical_package_name] = packages[package_name].get("index")
                     if packages[package_name].get("skip_resolver"):
                         is_constraint = False
                         skipped[package_name] = dep
                 elif index:
-                    index_lookup[package_name] = index
+                    index_lookup[canonical_package_name] = index
                 else:
-                    index_lookup[package_name] = project.get_default_index()["name"]
+                    index_lookup[canonical_package_name] = project.get_default_index()["name"]
             if install_req.markers:
                 markers_lookup[package_name] = install_req.markers
```

Comment:
Maybe we should use everywhere canonical_package_name instead of raw_package_name.
But i didn't start to do it because it would make affect on locking files => i'm a little bit afraid of it.

Reply:
Yeah you are right, I wish this is something I hadn't broke, but I think your patch makes sense as an iteration in the right direction.  I will think more about it.

Reply:
Can we apply my patch?

P.S. I sent it because at my team we have some issues with pipenv. And we can't migrate to the freshest version (we still use pretty old one)